src/pudl_archiver/archivers/epacems.py

```diff
@@ -86,5 +89,56 @@ async def get_year_quarter_resource(
             f"File no. {request_count}: Downloaded {year} Q{quarter} EPA CEMS hourly emissions data."
         )
         return ResourceInfo(
-            local_path=archive_path, partitions={"year_quarter": f"{year}q{quarter}"}
+            local_path=archive_path,
+            partitions={"year": year, "quarter": quarter},
+            layout=ZipLayout(filepaths=[filename]),
         )
+
+    def dataset_validate_archive(
+        self,
+        baseline_datapackage: DataPackage | None,
+        new_datapackage: DataPackage,
+        resources: dict[str, ResourceInfo],
+    ) -> list[validate.DatasetSpecificValidation]:
+        """Look for consecutive year quarter partitions in CEMS archive."""
+        year_quarters = [x.parts["year_quarter"] for x in new_datapackage.resources]
+        yq_split = [x.split("q") for x in year_quarters]
+        yq_df = pd.DataFrame(yq_split, columns=["year", "quarter"]).apply(
+            pd.to_numeric, errors="coerce"
+        )
+        newest_year = yq_df.year.max()
+        success = True
+        description = ""
+
+        # Test that there are no extraneous quarter values like q5
+        if not yq_df["quarter"].isin(range(1, 5)).all():
+            success = False
+            description = (
+                description
+                + f"""
+                f"The following years have bad quarters: \
+                {yq_df[yq_df['quarter'].isin(range(1,5))].year.unique().tolist()}. """
+            )
+        # Test that all old years have all four quarters
+        q_groups = yq_df[yq_df["year"] != newest_year].groupby("year").count()
+        if not q_groups.quarter.isin([4]).all():
+            success = False
+            description = (
+                description
+                + f"""
+                f"The following years do not have four quarters: \
+                    {q_groups[q_groups['quarter']!=4].index.tolist()}. """
+            )
+        # Test that the newest year has consecutive quarters
+        new_quarters = yq_df[yq_df["year"] == newest_year].quarter.unique().tolist()
+        if len(set(range(1, len(new_quarters) + 1)) - set(new_quarters)) != 0:
+            success = False
+            description = (
+                description + "There are missing quarters in the new year of data. "
+            )
```

Comment:
Do you mean like a logging output or a code comment?

Reply:
For other tests `description` is a static description of what the test does (e.g. "Check to ensure that CEMS data is consecutive") and the `notes` field offers descriptions of the specific test info (e.g., these quarters are missing, this file fails the conditions). I think keeping to this format is more consistent and will give more info to the person reviewing the summary .json about what the test is, since currently the description is blank if it passes.